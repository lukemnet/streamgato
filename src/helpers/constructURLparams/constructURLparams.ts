import { ShorthandValues } from 'types';

const constructURLparams = (params: ShorthandValues) =>
params && Object.keys(params).length > 0
    ? `?${Object.keys(params).map(key =>
        `${key}=${encodeURIComponent(params[key])}`)
      .join('&')}`
    : "";

export default constructURLparams;
