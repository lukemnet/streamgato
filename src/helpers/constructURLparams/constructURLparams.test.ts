import constructURLparams from './constructURLparams';

const testParams = {
  h: 1,
  m: 2,
  s: 3,
}

it('should construct URL params', () => {
  expect(constructURLparams(testParams)).toMatchSnapshot();
});


it('should return empty string for empty URL params', () => {
  expect(constructURLparams({})).toEqual("");
});