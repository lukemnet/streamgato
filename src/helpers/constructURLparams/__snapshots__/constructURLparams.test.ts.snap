// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should construct URL params 1`] = `"?h=1&m=2&s=3"`;
